ngraph/core/src/op/split.cpp

```diff
@@ -66,16 +62,34 @@ void op::v1::Split::validate_and_infer_types()
 
             NODE_VALIDATION_CHECK(this,
                                   dimension_at_axis % m_num_splits == 0,
-                                  "The input tensor's dimension pointed by the 'axis' parameter: ",
+                                  "Dimension of data input shape along 'axis': ",
                                   dimension_at_axis,
-                                  " has to be a multiple of the 'num_splits' attribute value: ",
+                                  " must be evenly divisible by 'num_splits' attribute value: ",
                                   m_num_splits);
 
             each_output_shape[axis] = dimension_at_axis / m_num_splits;
         }
         else
         {
-            each_output_shape[axis] = Dimension::dynamic();
+            const auto dim_interval_at_axis = data_ps[axis].get_interval();
+            NODE_VALIDATION_CHECK(
+                this,
+                dim_interval_at_axis.get_max_val() >= static_cast<int64_t>(m_num_splits),
+                "The interval maximum of the dimension for data input shape along 'axis' must be "
+                "greater or equal to 'num_splits' attribute. Got: ",
+                dim_interval_at_axis,
+                " and ",
+                m_num_splits);
+
+            auto dim_interval_at_axis_min =
+                static_cast<int64_t>(dim_interval_at_axis.get_min_val() * (1.0f / m_num_splits));
+            auto dim_interval_at_axis_max = dim_interval_at_axis.get_max_val();
+            if (dim_interval_at_axis.has_upper_bound())
+            {
+                dim_interval_at_axis_max =
+                    static_cast<int64_t>(dim_interval_at_axis_max * (1.0f / m_num_splits));
```

Comment:
num_splits could be equal to zero -- not from the specification but from the code I read.

Reply:
I added a node validation check on `m_num_splits` attribute.